internal/cli/atlas/search/nodes/spec_file.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodes
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/spf13/afero"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115006/admin"
+)
+
+type SpecFile []SpecFileEntry
+
+type SpecFileEntry struct {
+	InstanceSize string
+	NodeCount    int
+}
+
+// Load []atlasv2.ApiSearchDeploymentSpec from a given file.
+func LoadAPISearchDeploymentSpec(fs afero.Fs, filename string) ([]atlasv2.ApiSearchDeploymentSpec, error) {
+	spec := new(SpecFile)
```

Comment:
why not `spec := new([]atlasv2.ApiSearchDeploymentSpec)`?

Reply:
I've considered that. But if the api team decides to change field names on `atlasv2.ApiSearchDeploymentSpec` then the contract we have in our docs is broken.

Reply:
good point but as I mentioned in another comment we don't put the contract in the doc but rather to another page + this is the approach we use with the files. once the API changes, our E2E tests should fail and we should be able to detect this breaking change.

Reply:
got rid of the intermediate type